### PR TITLE
initial block storage version 1.0

### PR DESCRIPTION
**Pull request by HeinzOracle**
initial block storage version including
- Team Publications
- Useful Links
- Reusable Assets

### CODE REVIEW

File: cloud-infrastructure/storage/block-storage/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+We've designed our storage platform as an ideal complement to Oracle compute and networking services to support the highest performance requirements. OCI Block Volume uses the latest NVMe SSDs and provides nonblocking network connectivity to every host. Oracle delivers consistent, low-latency performance of up to 225 IOPS/GB to a maximum of 300,000 IOPS and 2,680 MB/sec of throughput per volume.
+
+Features
+
+- Cross-region asynchronous block volume replication
+- Consistent high performance
+- Dynamic performance scaling with auto-tuning of block volumes
+- Scales volumes from 50 GB to 32 TB
+- Attach volumes to reach 1 PB per compute instance
+- Less than half the price of AWS EBS
```
Comment (RafSzaf) [inline cloud-infrastructure/storage/block-storage/README.md:14]: Please remove direct reference to competition.

File: cloud-infrastructure/storage/block-storage/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+## Reusable Assets Overview
+
+- [Cloud Resilience](https://gitlab.com/hmielimo/cloud-resilience/-/blob/main/doc/cloud.resilience/README.md)
+  - All details regarding storage based resiliency incl. example backup & recover automation script
+- [Cloud Resilience by default](https://gitlab.com/hmielimo/cloud-resilience-by-default/)
+  - All details regarding storage based Server resiliency incl.
+    - Set up your Server with Resilience by default Using the Console
+    - Set up your Server with Resilience by default using CLI
+    - Performance-Based Auto-tune Use Case script
+- [Block Volume Service performance-based auto-tune - Cost comparison](https://gitlab.com/hmielimo/cloud-resilience-by-default/-/blob/main/doc/Block.Volume.Service.performance.based.auto.tune.Cost.comparison.xlsx)
```
Comment (RafSzaf) [inline cloud-infrastructure/storage/block-storage/README.md:66]: Please remove due to contained:
- Oracle pricing details
- Direct comparison vs AWS

**Review by RafSzaf — CHANGES_REQUESTED** [review]
In: cloud-infrastructure/storage/block-storage/README.md
14. Please remove direct reference to competition.
66. Please remove due to the below information contained in the excel file: Oracle pricing details, Direct comparison vs AWS.
Also add copyright notices at the top of each file where missing.
Thx,
Rafal

File: cloud-infrastructure/storage/block-storage/README.md (hunk 1 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/README.md]: done

File: cloud-infrastructure/storage/block-storage/README.md (hunk 2 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/README.md]: done

**Comment by AlexanderHodicke** [comment]
Hi in order to pass the CLA check you need to create a comment (like I'm writing right now) and just cope and paste

I have read the CLA Document and I hereby sign the CLA

Ill send you the agreement in Slack as the link above is broken

File: cloud-infrastructure/storage/block-storage/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# Block Storage
+
```
Comment (AlexanderHodicke) [inline cloud-infrastructure/storage/block-storage/README.md:2]: I would recommend putting this link under 'Useful Links'

File: cloud-infrastructure/storage/block-storage/asset/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Asset
+
```
Comment (AlexanderHodicke) [inline cloud-infrastructure/storage/block-storage/asset/README.md:2]: So the idea with the assets are to have a folder per document/file-based asset. 

.. / block-storage / how-to-guide-ABC / files / guide.doc
.. / block-storage / health-check / files / healthcheck.pdf

Each <asset name> folder has a README.md which is templatized as per my Confluence guide page and a LICENSE file. The actual asset binaries are under the files sub-folder.

File: cloud-infrastructure/storage/block-storage/asset/block.volume.backup.and.restore.txt [hunk 5]
```diff
@@ -0,0 +1,155 @@
+License
```
Comment (AlexanderHodicke) [inline cloud-infrastructure/storage/block-storage/asset/block.volume.backup.and.restore.txt:1]: asset folder structure seems a little off

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt [hunk 6]
```diff
@@ -0,0 +1,155 @@
+License
```
Comment (AlexanderHodicke) [inline cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:1]: @oheimburger code review please

File: cloud-infrastructure/storage/block-storage/asset/secure.storage.sh [hunk 7]
```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
```
Comment (AlexanderHodicke) [inline cloud-infrastructure/storage/block-storage/asset/secure.storage.sh:1]: @oheimburger  code review please

File: cloud-infrastructure/storage/block-storage/asset/trim.showcase.txt [hunk 8]
```diff
@@ -0,0 +1,198 @@
+TRIM showcase
```
Comment (AlexanderHodicke) [inline cloud-infrastructure/storage/block-storage/asset/trim.showcase.txt:1]: @oheimburger Code review please

**Comment by HeinzOracle** [comment]
> **CLA Assistant Lite bot:** Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you all sign our [Contributor License Agreement](https://github.com/oracledevrel/devrel-oca-mgmt/blob/main/oca-20210504.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.
> 
> _**I have read the CLA Document and I hereby sign the CLA**_
> 
> **1** out of **2** committers have signed the CLA.✅ @AlexanderHodicke❌ @HeinzOracleYou can retrigger this bot by commenting **recheck** in this Pull Request

I have read the CLA Document and I hereby sign the CLA

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt (hunk 6 above)
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:1]: Should this be run as a shell script? If so, start start with `#!/bin/bash`

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:1]: A a general rule: All comments should follow proper English grammar and punctuation, esp. start with a capital letter and end sentences with a period.

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+set needed variable
+{----------------------------------------------------------------------------------------------
+DEVICEoPATH1=/dev/oracleoci/oraclevdc
+DEVICEoPATH2=/dev/oracleoci/oraclevdc1
+MYoSOURCE1=/dev/sdb
+MYoSOURCE2=/dev/sdb1
+DATAoDIR=/mnt/MyBlockVolume.backup-test
+OBJECTDATAoDIR=/mnt/MyObjectStorage.backup-test
+BUCKEToNAME=backup-test
+MYoREGION=eu-frankfurt-1
```
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:36]: Make this a neutral value like `<region>`

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt (hunk 6 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:1]: no, the idea is to copy and paste code sections into a terminal

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+MYoNAMESPACE=***
```
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:37]: Add placeholders for `***`, like `<your_namespace>`.

File: cloud-infrastructure/storage/block-storage/asset/secure.storage.sh [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+# Update history:
+#
+# V 1.0.0 28.06.2023 initial version
+#
+
+# ---------------------------------------------------------------------------------------------------------------------------------------------
+# prepare environement (load functions)
+# ---------------------------------------------------------------------------------------------------------------------------------------------
+
+source functions0.sh
```
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/secure.storage.sh:21]: Where to find this file?

File: cloud-infrastructure/storage/block-storage/asset/secure.storage.sh [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+fi
+
+color_print "${MYcolor}" "${PF1} Rotation of Customer managed Key"
+if [ 1 -eq 1 ] ; then # Rotation of Customer managed Key
+
+: ' ---------------------------------------------------------------------------------------------------------------------------------------
+Periodically rotating keys limits the amount of data 
+encrypted or signed by one key version. If a key is 
+ever compromised, key rotation thus reduces the risk. 	https://docs.oracle.com/en-us/iaas/Content/KeyManagement/Concepts/keyoverview.htm#concepts
+kms management key-version create						https://docs.oracle.com/en-us/iaas/tools/oci-cli/latest/oci_cli_docs/cmdref/kms/management/key-version/create.html
```
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/secure.storage.sh:140]: Indentation may lead to unaligned display. Change TABs to SPACEs.

**Comment by HeinzOracle** [comment]
I have read the CLA Document and I hereby sign the CLA

File: cloud-infrastructure/storage/block-storage/asset/trim.showcase.txt
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/trim.showcase.txt:1]: This text file could be either a shell script, a set of script files or a markdown file.

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:1]: This text file could be either a shell script, a set of script files or a markdown file.

**Review by oheimburger — CHANGES_REQUESTED** [review]

File: cloud-infrastructure/storage/block-storage/README.md (hunk 3 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/README.md:2]: done

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt (hunk 9 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:36]: done

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt (hunk 10 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:37]: done

File: cloud-infrastructure/storage/block-storage/asset/secure.storage.sh (hunk 11 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/asset/secure.storage.sh:24]: just uploaded both needed files parameter1.sh and functions0.sh

File: cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/asset/block-volume-backup-and-restore.txt:1]: done

**Review by RafSzaf — DISMISSED** [review]

File: cloud-infrastructure/storage/block-storage/README.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+High-performance block storage at any scale
```
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/README.md:3]: End of sentence missing. Add a period ('.').

File: cloud-infrastructure/storage/block-storage/asset/README.md [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+OCI Block Storage assets.
```
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/README.md:3]: Change this to a link to '[OCI Block Storage assets](asset/README.md)'

File: cloud-infrastructure/storage/block-storage/asset/trim-showcase.md [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+
+### Set needed variables (*please adjust variables to your needs before past into your terminal*)
+~~~
+# ---------------------------------------------------------------------------------------------------------------------------------------------
+# CUSTOMER SPECIFIC VALUES - please update appropriate
+# ---------------------------------------------------------------------------------------------------------------------------------------------
+export TENENCY_OCID=<your_tenency_ocid>
+export USER_OCID=<your_user_ocid>
+export COMPARTMENT_OCID=<your_compartment_ocid>
+export FRANKFURT_SUBNET_OCID=<your_frankfurt_subnet_ocid>
```
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/trim-showcase.md:47]: Why FRANKFURT? Can this be region neutral?

File: cloud-infrastructure/storage/block-storage/asset/trim-showcase.md [hunk 16]
```diff
@@ -0,0 +1,199 @@
...
```
Comment (oheimburger) [inline cloud-infrastructure/storage/block-storage/asset/trim-showcase.md:129]: Again why FRANKFURT? Can this be region neutral?

**Review by oheimburger — CHANGES_REQUESTED** [review]

File: cloud-infrastructure/storage/block-storage/README.md (hunk 13 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/README.md]: done

File: cloud-infrastructure/storage/block-storage/asset/trim-showcase.md (hunk 15 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/asset/trim-showcase.md:47]: done

File: cloud-infrastructure/storage/block-storage/asset/trim-showcase.md (hunk 16 above)
Comment (HeinzOracle) [reply cloud-infrastructure/storage/block-storage/asset/trim-showcase.md:129]: done

**Review by oheimburger — APPROVED** [review]

**Pull request merged**